src/traces/sunburst/helpers.js

```diff
@@ -116,17 +109,78 @@ exports.determineInsideTextFont = function(trace, pt, layoutFont) {
         customColor = Lib.castOption(trace._input, ptNumber, 'textfont.color');
     }
 
-    var family = Lib.castOption(trace, ptNumber, 'insidetextfont.family') ||
-        Lib.castOption(trace, ptNumber, 'textfont.family') ||
-        layoutFont.family;
-
-    var size = Lib.castOption(trace, ptNumber, 'insidetextfont.size') ||
-        Lib.castOption(trace, ptNumber, 'textfont.size') ||
-        layoutFont.size;
-
     return {
         color: customColor || Color.contrast(cdi.color),
-        family: family,
-        size: size
+        family: exports.getInsideTextFontKey('family', cont || trace, pt, layoutFont),
+        size: exports.getInsideTextFontKey('size', cont || trace, pt, layoutFont)
     };
+}
+
+exports.getInsideTextFontKey = function(keyStr, trace, pt, layoutFont) {
+    var ptNumber = pt.data.data.i;
+
+    return (
+        Lib.castOption(trace, ptNumber, 'insidetextfont.' + keyStr) ||
+        Lib.castOption(trace, ptNumber, 'textfont.' + keyStr) ||
+        layoutFont.size
+    );
+};
+
+exports.getOutsideTextFontKey = function(keyStr, trace, pt, layoutFont) {
+    var ptNumber = pt.data.data.i;
+
+    return (
+        Lib.castOption(trace, ptNumber, 'outsidetextfont.' + keyStr) ||
+        Lib.castOption(trace, ptNumber, 'textfont.' + keyStr) ||
+        layoutFont.size
+    );
+};
+
+exports.isOutsideText = function(trace, pt) {
+    return !trace._hasColorscale && exports.isHierarchyRoot(pt);
+};
+
+exports.determineTextFont = function(trace, pt, layoutFont, cont) {
+    return exports.isOutsideText(trace, pt) ?
+        determineOutsideTextFont(trace, pt, layoutFont) :
+        determineInsideTextFont(trace, pt, layoutFont, cont);
+};
+
+exports.hasTransition = function(transitionOpts) {
+    // We could optimize hasTransition per trace,
+    // as sunburst & treemap have no cross-trace logic!
+    return !!(transitionOpts && transitionOpts.duration > 0);
+};
+
+exports.strTransform = function(d) {
+    return getTransform({
+        textX: d.transform.textX,
+        textY: d.transform.textY,
+        targetX: d.transform.targetX,
+        targetY: d.transform.targetY,
+        scale: d.transform.scale,
+        rotate: d.transform.rotate
+    });
+};
+
+exports.getMaxDepth = function(trace) {
+    return trace.maxdepth >= 0 ? trace.maxdepth : Infinity;
+};
+
+exports.isHeader = function(pt, trace) { // it is only used in treemap.
+    return !(exports.isLeaf(pt) || pt.depth === trace._maxDepth - 1);
+};
+
+exports.getLabelStr = function(label) {
+    return has(label) ? label.split('<br>').join(' ') : '';
+};
+
+exports.getLabelString = function(label) { // used in hover to reference to the "root"
+    var str = exports.getLabelStr(label);
+    return str ? str : '"root"';
```

Comment:
> When hovering over an empty "root"
> 
> ![Peek 2019-09-12 17-53](https://user-images.githubusercontent.com/6675409/64823759-461ab480-d586-11e9-97fa-45a645c3027f.gif)
> 
> I would simply not show any hover labels nor hover effect.

For `sunburst` because the elements are arranged around a point it is clear how they grouped e.g. when there are multiple of them. But when having multiple `treemap` traces side by side,
I think this is a pretty useful effect. 
Having said that we should probably drop that 100% of self thing.

Reply:
So, you're thinking of keeping the hover effect, but dropping the hover label when hovering over an empty root sector?

Reply:
... I'd be :ok_hand: with that!

Reply:
> So, you're thinking of keeping the hover effect, but dropping the hover label when hovering over an empty root sector?

Not 100%. IMO displaying hover on root is useful. 
I was thinking of not displaying the `percent of root` in the hover; because the root is always 100% of itself.

Reply:
Oh, so the hover label would just show "root" in this case? That's a very nice solution also :ok_hand:

Reply:
Current behaviour with `hoverinfo: 'all'`:

```js
Plotly.newPlot(gd, [{
   type: 'treemap',
   labels: [ 'A', 'B', 'b'],
   parents: ['', '', 'B'],
   hoverinfo: 'all'
}], {

  shapes: [{
    x0: 0, x1: 1, xref: 'paper',
    y0: 0, y1: 1, yref: 'paper'
  }],
  width: 500,
  height: 500
})
```

gives:

![Peek 2019-09-20 11-25](https://user-images.githubusercontent.com/6675409/65339006-57bd1700-db99-11e9-8020-7359b6f9c3fe.gif)

which is _ok_ to me. Especially now that `hoverinfo: 'label+text+value+name'`  by default. So I'll resolve this comment.